apps/web/pages/api/sendDailyVideoRecordingEmail.ts

```diff
@@ -0,0 +1,137 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { z } from "zod";
+
+import { getDownloadLinkOfCalVideoByRecordingId } from "@calcom/core/videoClient";
+import { sendDailyVideoRecordingEmails } from "@calcom/emails";
+import { defaultHandler } from "@calcom/lib/server";
+import { getTranslation } from "@calcom/lib/server/i18n";
+import prisma, { bookingMinimalSelect } from "@calcom/prisma";
+import type { CalendarEvent } from "@calcom/types/Calendar";
+
+const schema = z.object({
+  recordingId: z.string(),
+  bookingUID: z.string(),
+});
+
+const downloadLinkSchema = z.object({
+  download_link: z.string(),
+});
+
+async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (!process.env.SENDGRID_API_KEY || !process.env.SENDGRID_EMAIL) {
+    return res.status(405).json({ message: "No SendGrid API key or email" });
+  }
+  const response = schema.safeParse(JSON.parse(req.body));
+  console.log("REQ", response);
+
+  if (!response.success) {
+    return res.status(400).send({
+      message: "Invalid Payload",
+    });
+  }
+
+  const { recordingId, bookingUID } = response.data;
+
+  try {
+    const booking = await prisma.booking.findFirst({
+      where: {
+        uid: bookingUID,
```

Comment:
We must check for logged in user ID here otherwise, the endpoint being public, anyone can send an email for any booking